### ISSUE TITLE
Start-Job does not work with SDK image 6.0.100

### ISSUE THREAD

**Issue by ThomasPiskol** [dotnet/dotnet-docker#3302]
### Describe the Bug
❗ As reported by @cboulva99 below, this issue is not only affecting the Windows images but also the Linux images.

It's not possible to use the `Start-Job` cmdlet in the latest Windows Server Core 2019 amd64 SDK image. Using this cmdlet fails with the following error message:

```shell
Start-Job: The pwsh executable cannot be found at "C:\Program Files\powershell\.store\powershell.windows.x64\7.2.0\powershell.windows.x64\7.2.0\tools\net6.0\any\pwsh.exe".
Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenar
ios.
```

### Steps to Reproduce

```shell
docker run -it --rm mcr.microsoft.com/dotnet/sdk:6.0.100-windowsservercore-ltsc2019 pwsh "-c" "Start-Job" "-ScriptBlock" "{ Write-Host 'Hello Block' }"
```

Output:
```shell
Start-Job: The pwsh executable cannot be found at "C:\Program Files\powershell\.store\powershell.windows.x64\7.2.0\powershell.windows.x64\7.2.0\tools\net6.0\any\pwsh.exe".
Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenar
ios.
```

With previous versions (e.g.  **5.0.403-windowsservercore-ltsc2019**) it worked as expected:

```shell
docker run -it --rm mcr.microsoft.com/dotnet/sdk:5.0.403-windowsservercore-ltsc2019 pwsh "-c" "Start-Job" "-ScriptBlock" "{ Write-Host 'Hello Block' }"
```
Output:
```shell
Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
--     ----            -------------   -----         -----------     --------             -------
1      Job1            BackgroundJob   Running       True            localhost             Write-Host 'Hello Block…
```

### Other Information

This issue happens all the time, there's no workaround.

I've noticed that `pwsh.exe` is actually missing in the above mentioned folder `C:\Program Files\powershell\.store\powershell.windows.x64\7.2.0\powershell.windows.x64\7.2.0\tools\net6.0\any`.

In **5.0.403-windowsservercore-ltsc2019** `pwsh.exe` is present in the folder `C:\Program Files\powershell\.store\powershell.windows.x64\7.1.5\powershell.windows.x64\7.1.5\tools\net5.0\any`

### Output of `docker version`

```txt
Client:
 Cloud integration: v1.0.20
 Version:           20.10.10
 API version:       1.41
 Go version:        go1.16.9
 Git commit:        b485636
 Built:             Mon Oct 25 07:47:53 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.10
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.16.9
  Git commit:       e2f740d
  Built:            Mon Oct 25 07:43:13 2021
  OS/Arch:          windows/amd64
  Experimental:     false
```

### Output of `docker info`

```txt
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.6.3)
  compose: Docker Compose (Docker Inc., v2.1.1)
  scan: Docker Scan (Docker Inc., 0.9.0)

Server:
 Containers: 3
  Running: 0
  Paused: 0
  Stopped: 3
 Images: 53
 Server Version: 20.10.10
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 19043 (19041.1.amd64fre.vb_release.191206-1406)
 Operating System: Windows 10 Enterprise Version 2009 (OS Build 19043.1348)
 OSType: windows
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.88GiB
 Name: md1r5gfc
 ID: LQ5H:HQ63:QEW4:N3MM:WJ4T:5AXU:UCLH:IFQT:KWD6:XPBM:J4UR:HOQR
 Docker Root Dir: C:\ProgramData\Docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Comment by mthalman** [comment]
PTAL @adityapatwardhan

**Comment by cboulva99** [comment]
Hello!

We are having the same problem with the focal version of the image (**mcr.microsoft.com/dotnet/sdk:6.0-focal**)

As a symptom, the **$PSHOME** variable seems to be misconfigured, as it is pointing to:
`/usr/share/powershell/.store/powershell.linux.x64/7.2.0/powershell.linux.x64/7.2.0/tools/net6.0/any/pwsh`

But this path does not exist.
So, while pwsh can be launched, it cannot in turn start any command with the 
`& xxxxx.ps1 command`

**Comment by adityapatwardhan** [comment]
Having a look at this. In most cases, as a workaround `Start-ThreadJob` can be used.

**Comment by ClaudiuSchuster** [comment]
@adityapatwardhan  The problem @ThomasPiskol is facing is that in the first step he is registering new PSRepositories which can only be used in a new session (what in this case is not a workaround) or a script-block executed by start-job, but not a scriptblock executed by start-threadjob.

**Comment by ThomasPiskol** [comment]
@adityapatwardhan Rewriting all scripts to replace `Start-Job` with `Start-ThreadJob` doesn't seem to be a viable solution, unless [this text](https://docs.microsoft.com/en-us/powershell/scripting/overview?view=powershell-7.2) is rephrased 😉:

> PowerShell is a cross-platform task automation solution made up of a command-line shell, a scripting language, and a configuration management framework. PowerShell runs on Windows, Linux, and macOS **except for the official .NET SDK docker images**.

But as @cboulva99 and @ClaudiuSchuster mentioned, there are other use cases which are broken right now.
❗ Thanks to @cboulva99 we know now that this is issue also happens on Linux too.

Here's another example, which is a little bit more complicated to reproduced, but here we go:

#### 1. Create a new Dockerfile with the following content 👇
Windows:
```Dockerfile
# escape=`
FROM mcr.microsoft.com/dotnet/sdk:6.0.100-windowsservercore-ltsc2019
SHELL ["pwsh","-command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]

RUN & C:\Windows\System32\whoami.exe `
    & cmd.exe /c echo What will happen...
```

Debian:
```Dockerfile
# escape=`
FROM mcr.microsoft.com/dotnet/sdk:6.0.100-bullseye-slim
SHELL ["pwsh","-command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]

RUN & whoami `
    & bash -c "echo 'What will happen...'"
```


#### 2. Try to build the image
```shell
docker build . -t test
```

#### 3. Result:
Windows:
```shell
Sending build context to Docker daemon  2.048kB
Step 1/3 : FROM mcr.microsoft.com/dotnet/sdk:6.0.100-windowsservercore-ltsc2019
 ---> 2f378c936d99
Step 2/3 : SHELL ["pwsh","-command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
 ---> Using cache
 ---> bf6b11be7f7b
Step 3/3 : RUN & C:\Windows\System32\whoami.exe     & cmd.exe /c echo What will happen...
 ---> Running in 192e677d83d1
Start-Job: The pwsh executable cannot be found at "C:\Program Files\powershell\.store\powershell.windows.x64\7.2.0\powershell.windows.x64\7.2.0\tools\net6.0\any\pwsh.exe".
Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
The command 'pwsh -command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; & C:\Windows\System32\whoami.exe     & cmd.exe /c echo What will happen...' returned a non-zero code: 1
```

Debian
```shell
[+] Building 1.5s (5/5) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                       0.0s
 => => transferring dockerfile: 487B                                                                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                            0.0s
 => [internal] load metadata for mcr.microsoft.com/dotnet/sdk:6.0.100-bullseye-slim                                                                                                                        0.0s
 => CACHED [1/2] FROM mcr.microsoft.com/dotnet/sdk:6.0.100-bullseye-slim                                                                                                                                   0.0s
 => ERROR [2/2] RUN & whoami     & bash -c "echo 'What will happen...'"                                                                                                                                    1.4s
------
 > [2/2] RUN & whoami     & bash -c "echo 'What will happen...'":
#5 1.311 Start-Job: The pwsh executable cannot be found at "/usr/share/powershell/.store/powershell.linux.x64/7.2.0/powershell.linux.x64/7.2.0/tools/net6.0/any/pwsh".
#5 1.311 Note that 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications. Instead, usage of the 'ThreadJob' module is recommended in such scenarios.
------
executor failed running [pwsh -command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; & whoami     & bash -c "echo 'What will happen...'"]: exit code: 1
```


Like @cboulva99 already mentioned, it seems to be related to the fact, that the `$PSHOME` folder doesn't contain `pwsh.exe`.

#### 4. Workaround
Modifying the sample Dockerfile from above and adding a manual copy step for `pwsh.exe` seems to work. But I'm not sure what else is broken:

```Dockerfile
# escape=`
FROM mcr.microsoft.com/dotnet/sdk:6.0.100-windowsservercore-ltsc2019
SHELL ["pwsh","-command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]

# Workaround for https://github.com/dotnet/dotnet-docker/issues/3302
RUN Copy-Item (Get-Command pwsh).Source $PSHOME

RUN & C:\Windows\System32\whoami.exe `
    & cmd.exe /c echo What will happen...
```
 this seems to be successful:

```shell
docker build . -t test
Sending build context to Docker daemon  3.584kB
Step 1/4 : FROM mcr.microsoft.com/dotnet/sdk:6.0.100-windowsservercore-ltsc2019
 ---> 2f378c936d99
Step 2/4 : SHELL ["pwsh","-command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
 ---> Using cache
 ---> bf6b11be7f7b
Step 3/4 : RUN Copy-Item (Get-Command pwsh).Source $PSHOME
 ---> Running in 08d086ae69e1
Removing intermediate container 08d086ae69e1
 ---> b2dc511a13d0
Step 4/4 : RUN & C:\Windows\System32\whoami.exe     & cmd.exe /c echo What will happen...
 ---> Running in d88514952917

Id     Name            PSJobTypeName   State         HasMoreData     Location
--     ----            -------------   -----         -----------     --------
1      Job1            BackgroundJob   Running       True            localhost
What will happen...

Removing intermediate container d88514952917
 ---> 95e8913fae8a
Successfully built 95e8913fae8a
Successfully tagged test:latest
```

Even if the solution is this simple, it should be fixed in the SDK image. Otherwise a lot of people will run into strange issues.

**Comment by adityapatwardhan** [comment]
I am sorry if I was not clear. We are committed to fix this issue ASAP. I have understood the problem and am already working on a fix. We will try to release this with the next update.

**Comment by cboulva99** [comment]
Hello,

Since the Powershell image was fixed by @adityapatwardhan, can we expect to have a new sdk image soon? I am looking in the nightly builds or official builds and I don't see any new image containing the fix? Or maybe we must use a specific tag image, instead of _mcr.microsoft.com/dotnet/nightly/sdk:6.0_?

Thank your for your help!

**Comment by mthalman** [comment]
@cboulva99 - The nightly tag does include the new version:

```console
❯ docker pull mcr.microsoft.com/dotnet/nightly/sdk:6.0
6.0: Pulling from dotnet/nightly/sdk
e5ae68f74026: Already exists
dbb945cbc911: Pull complete
bb3474c336e4: Pull complete
f60e5bbd1977: Pull complete
02d5470fdb5a: Pull complete
bf02a54bb298: Pull complete
49f599ce127d: Pull complete
be270ee32b01: Pull complete
Digest: sha256:fd74d594d9a71979989388097de461a1f2a9dc58f4e951be5feb15bb08f11c77
Status: Downloaded newer image for mcr.microsoft.com/dotnet/nightly/sdk:6.0
mcr.microsoft.com/dotnet/nightly/sdk:6.0

❯ docker run --rm mcr.microsoft.com/dotnet/nightly/sdk:6.0 pwsh --version
PowerShell 7.2.1
```

Are you not seeing the issue fixed with that version? If not, can you provide repro steps? cc @adityapatwardhan 

The official images will be published tomorrow as part of the monthly Patch Tuesday release.

**Comment by cboulva99** [comment]
Thank you @mthalman ,


I messed up in my Docker file, i had two references to the microsoft registry, one with the nightly build reference, one without ;)
My bad, thank you for the information (anyway I'll change by my reference to the official after tomorrow's release is ready).

Thanks!

**Comment by mthalman** [comment]
This has been fixed as part of the 6.0.101 SDK release on Dec. 14.

**Comment by cboulva99** [comment]
Hello team, @mthalman  @adityapatwardhan , before starting a new completely unrelated issue, I figured I'd add a message to this existing one, since I'm having the problem once again using the latest **alpine3.18 distribution of .net 7.0 sdk package** and it might be directly related.

Here it goes: I'm using the following image: 
`mcr.microsoft.com/dotnet/sdk:7.0-alpine3.18`

The image seems to have pwsh installed, as if I run:
`kubectl exec [..] -it  -- pwsh`, I get:
```
PowerShell 7.3.4

   A new PowerShell stable release is available: v7.3.5
   Upgrade now, or check out the release page at:
     https://aka.ms/PowerShell-Release?tag=v7.3.5
```

Now, here comes the issues:
Once I'm in the shell, 
* every attempt to run a command using & or Start-Job directly fails.
* Running the pwsh command itself ends the following way:
```
ResourceUnavailable: Program 'pwsh' failed to run: An error occurred trying to start process '/usr/share/powershell/.store/powershell.linux.alpine/7.3.4/powershell.linux.alpine/7.3.4/tools/net7.0/any/pwsh' with working directory '/docker/bin'. No such file or directoryAt line:1 char:1
+ pwsh
+ ~~~~
```

So basically, I cannot launch any background task NOR a new instance of pwsh using this docker image.

Thank you in advance for your time, and I hope adding to the existing issue will not cause you any headaches!

Regards,

**Comment by mthalman** [comment]
@cboulva99 - Please log a separate issue. Please provide the simplest set of repro steps you can find. For example, this issue had a repro step of `docker run -it --rm <image-name> pwsh "-c" "Start-Job" "-ScriptBlock" "{ Write-Host 'Hello Block' }"`. That command runs fine with `mcr.microsoft.com/dotnet/sdk:7.0-alpine3.18`, btw.
